File: Programmers/키패드 누르기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

string solution(vector<int> numbers, string hand) {
    string answer = "";
    int cl = 10;
    int cr = 12;
    int tmpL = 0;
    int tmpR = 0;
    for(int i = 0 ; i < numbers.size(); i++)
    {
        if(numbers[i]==0) numbers[i] = 11;
        if(numbers[i] % 3 == 1)
        {
            answer += 'L';
            cl = numbers[i];
        }
        else if (numbers[i] % 3 == 0)
        {
            answer += 'R';
            cr = numbers[i];
        }
        else
        {
            tmpL = abs(cl - numbers[i]);
            tmpR = abs(cr - numbers[i]);
            tmpL = tmpL / 3 + tmpL % 3;
            tmpR = tmpR / 3 + tmpR % 3;
            if(tmpL==tmpR && hand == "left")
            {
                answer += 'L';
                cl = numbers[i];
            }
            else if(tmpL==tmpR && hand == "right")
            {
                answer += 'R';
                cr = numbers[i];
            }
            else if(tmpL<tmpR)
            {
                answer += 'L'; 
                cl = numbers[i];
            }
            else
            {
                answer += 'R';
                cr = numbers[i];
            }
        }
    }
    return answer;
}

File: Programmers/키패드 누르기.java
class Solution {
	public String solution(int[] numbers, String hand) {
		String answer = "";
		int L = 10;
		int R = 12;
		for (int i = 0; i < numbers.length; i++) {
			if (numbers[i] == 0)
				numbers[i] = 11;
			int Le = Math.abs(numbers[i] - L) / 3 + Math.abs(numbers[i] - L) % 3;
			int Ri = Math.abs(numbers[i] - R) / 3 + Math.abs(numbers[i] - R) % 3;
			if (numbers[i] % 3 == 1) {
				L = numbers[i];
				answer += 'L';
			} else if (numbers[i] % 3 == 0) {
				R = numbers[i];
				answer += 'R';
			} else if (Le == Ri && hand.equals("left")) {
				L = numbers[i];
				answer += 'L';
			} else if (Le == Ri && hand.equals("right")) {
				R = numbers[i];
				answer += 'R';
			} else if (Le < Ri) {
				L = numbers[i];
				answer += 'L';
			} else if (Le > Ri) {
				R = numbers[i];
				answer += 'R';
			}
		}
		return answer;
	}
}